### PR TITLE
fix: bff skippable step handling

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** fixes handling of skipped steps in new ORA experience.

**What changed?**

- [Updates "has reached" check to account for skipped steps](https://github.com/openedx/edx-ora2/commit/4a7455a72bbcda44e472379c26dfe3121799b7b8)
- [Get next incomplete step instead of naively pulling from workflow status](https://github.com/openedx/edx-ora2/commit/ddcc1b4050b9cc5ce09c3e6a600682c2b7344b9f). Since new ORA experience uses "serial" instead of "parallel" progression, this correctly progresses us through the peer step when there are steps after it instead of skipping to the next non-peer step.

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [x] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Testing Instructions**

1. Create an ORA with a step after the peer step.
2. Complete the step before peer.
3. Verify that we get progressed to the peer step instead of skipping to the step after peer.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Review by jansenk — APPROVED** [review]
This seems reasonable to me
